src/System.Collections.Concurrent/tests/ProducerConsumerCollectionTests.cs

```diff
@@ -761,8 +768,11 @@ public void ManyConcurrentAddsTakes_CollectionRemainsConsistent(int operations)
                 // avoid 0 as default(T), to detect accidentally reading a default value
                 for (int i = 1; i < (1 + operations); i++)
                 {
-                    Assert.True(c.TryAdd(i));
-                    added.Add(i);
+                    if (c.Count < MaxConcurrentCollectionSize)
+                    {
+                        Assert.True(c.TryAdd(i));
+                        added.Add(i);
+                    }
```

Comment:
I'd prefer to see this fixed differently.  The goal of the test is to stress concurrent TryAdd/TryTake operations.  Currently it does that by having one thread just adding, another thread just taking, and another thread both adding and taking.  It'd be fine to simply have the add thread fall back to doing a take instead of an add if it hits the limit.

But there's separately another issue with the fix.  `c.Count` on concurrent collections is an expensive operation, and by doing the check on every iteration, it's slowing down the test and likely minimizing the actual amount of contention between TryAdd and TryTake.  Further, on some collections (like ConcurrentQueue), it can actually cause a change in how the queue operates, e.g. https://github.com/dotnet/corefx/blob/master/src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentQueue.cs#L356-L358 where it needs to freeze the segments in the queue in order to be able to get a stable snapshot count, which then influences how TryAdd works, forcing it to store into new segments.